component/async/kafka/simple/simple.go

```diff
@@ -326,6 +342,52 @@ func (c *consumer) partitionsSinceDuration(ctx context.Context) ([]sarama.Partit
 	return pcs, nil
 }
 
+func (c *consumer) partitionsSinceTimestamp(_ context.Context) ([]sarama.PartitionConsumer, error) {
+	client, err := sarama.NewClient(c.config.Brokers, c.config.SaramaConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create client: %w", err)
+	}
+
+	consumer, err := sarama.NewConsumerFromClient(client)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create simple consumer: %w", err)
+	}
+	c.ms = consumer
+
+	partitions, err := c.ms.Partitions(c.topic)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get partitions: %w", err)
+	}
+
+	pcs := make([]sarama.PartitionConsumer, len(partitions))
+
+	ts := c.config.TimestampOffset
+	c.startingOffsets = make(map[int32]int64)
```

Comment:
A very minor suggestion, just like a slice, to preallocate the hashtable.

```suggestion
	c.startingOffsets = make(map[int32]int64, len(partitions))
```